flytekit/annotated/python_function_task.py

```diff
@@ -167,9 +174,17 @@ def __init__(
             **kwargs,
         )
         self._task_function = task_function
+        self._execution_mode = execution_mode if execution_mode is not None else self.ExecutionMode.DEFAULT
+
+    @property
+    def execution_mode(self) -> ExecutionMode:
+        return self._execution_mode
 
     def execute(self, **kwargs) -> Any:
-        return self._task_function(**kwargs)
+        if self.execution_mode == self.ExecutionMode.DEFAULT:
+            return self._task_function(**kwargs)
+        elif self.execution_mode == self.ExecutionMode.DYNAMIC:
```

Comment:
is this a new execution mode?

Reply:
ohh yes. should i call it something else?